### PR TITLE
[WIP]Speed up parquet reading with Java Vector API

### PR DESCRIPTION
**Pull request by jiangjiguang**
### What changes were proposed in this pull request?
Parquet has supported vector read speed up with this PR https://github.com/apache/parquet-mr/pull/1011
The performance gain is 4x ~ 8x according to the parquet microbenchmark
TPC-H(SF100) Q6 has 11% performance increase with Apache Spark integrating parquet vector optimization

### Why are the changes needed?
This PR used to support parquet vector optimization

### Does this PR introduce _any_ user-facing change?
Add configuration spark.sql.parquet.vector512.read.enabled, If true and CPU contains avx512vbmi & avx512_vbmi2 instruction set, parquet decodes using Java Vector API. For Intel CPU, Ice Lake or newer contains the required instruction set.

### How was this patch tested?
For the test case, there are some problems to fix:
1. It is necessary to Parquet-mr community release new java version to use the parquet vector optimization.
2. Parquet Vector optimization does not release default, so users have to build parquet with mvn clean install -P vector-plugins manually to get the parquet-encoding-vector-{VERSION}.jar and put it on the {SPARK_HOME}/jars path
3. github doesn't support select runners with specific instruction set. So it is impossible (a self-hosted runner can do it) to verify the optimization on github runners machine.

### CODE REVIEW

**Comment by jiangjiguang** [comment]
@LuciferYang @wangyum @frankliee Since parquet-mr has released 1.13.0, So I resubmit the PR.

**Comment by jiangjiguang** [comment]
@frankliee Sorry for delay. The PR only supports AVX512, does not support AVX256.
Your question "Do we need to create SparkContext in static code ?"  because I want to get the SQL configuration sql.parquet.vector512.read.enabled

**Comment by jiangjiguang** [comment]
@LuciferYang @wangyum @frankliee I have added a benchmark.

This is the result:
```
Java HotSpot(TM) 64-Bit Server VM 17.0.5+9-LTS-191 on Linux 5.15.0-60-generic
Intel(R) Xeon(R) Platinum 8358P CPU @ 2.60GHz
Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Without Java Vector API                            4696           4802          89         21.3          47.0       1.0X
With Java Vector API                               3742           3927         230         26.7          37.4       1.3X
```

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java [hunk 1]
```diff
@@ -928,11 +956,35 @@ private boolean readNextGroup() {
           }
           currentBufferIdx = 0;
           int valueIndex = 0;
-          while (valueIndex < this.currentCount) {
-            // values are bit packed 8 at a time, so reading bitWidth will always work
-            ByteBuffer buffer = in.slice(bitWidth);
-            this.packer.unpack8Values(buffer, buffer.position(), this.currentBuffer, valueIndex);
-            valueIndex += 8;
+          if (vector512Support && SQLConf.get().parquetVector512Read()) {
```
Comment (LuciferYang) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java:959]: It seems that there is currently no corresponding UTs, and if not all GA machines can be used to check this part, how can we continue to ensure its robustness?

**Pull request closed without merging**